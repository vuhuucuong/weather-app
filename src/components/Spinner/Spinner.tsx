import React from 'react'
import cx from 'classnames'

interface SpinnerProps {
  className?: string
  size?: number
}

const Spinner: React.FC<SpinnerProps> = ({ className, size = 20 }) => {
  return (
    <div
      style={{ width: size, height: size }}
      className={cx('spinner-border', className)}
      role="status"
    >
      <span className="visually-hidden">Loading...</span>
    </div>
  )
}

export default Spinner
