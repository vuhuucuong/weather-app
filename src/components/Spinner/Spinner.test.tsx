import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Spinner from './Spinner'

it('should match snapshot', () => {
  expect(renderer.create(<Spinner />).toJSON()).toMatchSnapshot()
  expect(
    renderer.create(<Spinner className="mb-0" size={10} />).toJSON()
  ).toMatchSnapshot()
})

it('renders without crashing', () => {
  render(<Spinner className="mb-0" size={10} />)
})
