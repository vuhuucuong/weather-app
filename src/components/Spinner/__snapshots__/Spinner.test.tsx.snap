// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="spinner-border"
  role="status"
  style={
    Object {
      "height": 20,
      "width": 20,
    }
  }
>
  <span
    className="visually-hidden"
  >
    Loading...
  </span>
</div>
`;

exports[`should match snapshot 2`] = `
<div
  className="spinner-border mb-0"
  role="status"
  style={
    Object {
      "height": 10,
      "width": 10,
    }
  }
>
  <span
    className="visually-hidden"
  >
    Loading...
  </span>
</div>
`;
